tests/test-derivation.nix

```diff
@@ -9,32 +14,51 @@ let
       dontRun ? false,
       ...
     }:
+    let
+      testList =
+        if builtins.isList nvim then
+          nvim
+        else
+          [
+            {
+              derivation = nvim;
+              inherit name dontRun;
+            }
+          ];
+    in
     pkgs.stdenv.mkDerivation {
       inherit name;
 
-      nativeBuildInputs = [
-        nvim
-        pkgs.docker-client
-      ];
+      nativeBuildInputs = [ pkgs.docker-client ];
 
       dontUnpack = true;
       # We need to set HOME because neovim will try to create some files
       #
       # Because neovim does not return an exitcode when quitting we need to check if there are
       # errors on stderr
-      buildPhase =
-        if !dontRun then
-          ''
-            mkdir -p .cache/nvim
+      buildPhase = lib.optionalString (!dontRun) (
+        ''
+          mkdir -p .cache/nvim
+        ''
+        + lib.concatStringsSep "\n" (
+          builtins.map (
+            {
+              derivation,
+              name,
+              dontRun ? false,
+            }:
+            lib.optionalString (!dontRun) ''
+              echo "Running test for ${name}"
 
-            output=$(HOME=$(realpath .) nvim -mn --headless "+q" 2>&1 >/dev/null)
-            if [[ -n $output ]]; then
-            	echo "ERROR: $output"
-              exit 1
-            fi
-          ''
-        else
-          '''';
+              output=$(HOME=$(realpath .) ${lib.getExe derivation} -mn --headless "+q" 2>&1 >/dev/null)
+              if [[ -n $output ]]; then
+                echo "ERROR: $output"
+                exit 1
```

Comment:
Maybe in another PR, could we collect errors and print them all instead of exiting immediately?

Reply:
I'll do that in another PR